src/Jackalope/Node.php

```diff
@@ -302,7 +310,9 @@ private function parseData($rawData, $update, $keepChanges = false)
                 $this->nodes = $nodesInBackend;
                 $this->originalNodesOrder = null;
             }
-            foreach ($oldProperties as $name => $property) {
+            foreach ($oldPropertyValues as $name => $propertyValues) {
+                // TODO needs an update
+                $property = $this->loadProperty($name);
```

Comment:
TBH, I don't really know what's happening here. But looks like you are right, so I can't use the `loadProperties` method here... What would be the easiest way to achieve that? Would it be enough to only use the nodes that have already been instantiated before the call of `parseData`?